ecl/eclccserver/eclccserver.cpp

```diff
@@ -464,7 +464,30 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
 #ifdef _CONTAINERIZED
         if (!globals->getPropBool("@useChildProcesses", false) && !globals->hasProp("@workunit"))
         {
-            runK8sJob("eclccserver", wuid, wuid, globals->getPropBool("@deleteJobs", true));
+            try
+            {
+                runK8sJob("compile", wuid, wuid, globals->getPropBool("@deleteJobs", true));
+            }
+            catch (IException *E)
+            {
+                Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+                workunit.setown(factory->updateWorkUnit(wuid.get()));
+                if (workunit)
+                {
+                    if (workunit->aborting() || workunit->getState()==WUStateAborted)
+                        workunit->setState(WUStateAborted);
+                    else
+                    {
+                        StringBuffer msg;
+                        E->errorMessage(msg);
+                        addExceptionToWorkunit(workunit, SeverityError, "eclccserver", E->errorCode(), msg.str(), NULL, 0, 0, 0);
+                        workunit->setState(WUStateFailed);
+                    }
+                }
+                E->Release();
+                workunit->commit();
+                workunit.clear();
```

Comment:
minor: don't really need the above 2 lines.

Reply:
Again, copying style from elsewhere - and I think the commit MAY be needed (I can't remember the semantics!)

Reply:
Again, code elsewhere > 9 years old.  I think it is implicit in the destuctor for the workunit.

Reply:
I may not need the commit, but I think I need the clear (it's a member of the threadpool object, not a local).

Reply:
Looking around the code, committing before releasing seems to be the common pattern.

There is a commit called for the destructor of a CLockedWorkunit - but the commit may fail and throw an exception which is not necessarily desirable inside a destructor.

I'd prefer to have this code match the paradigm used elsewhere in this file (and the rest of the codebase). Removing unwanted calls to commit() should be a separate issue (and a very low-priority one!)